### ISSUE TITLE
bug: Ion-back-button don't work when change ion-tabs in ionic v6

### ISSUE THREAD

**Issue by nguyentichduy2k** [ionic-team/ionic-framework#24904]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

- ion-back-button only changes the path, not the view when i changing ion-tabs
- It's don't work on version ionic 6

### Expected Behavior

- ion-back-button can change the path and view when i changing ion-tabs 
- It's work on version ionic ^5.4

### Steps to Reproduce

I have an example on codesandbox: [https://codesandbox.io/s/example-router-bug-91320](url)
1. Click on change router button
2. Click to change tabs 
3. Click back button => it don't work here
4. Change version of ionic in `package.json `     
```
  "@ionic/vue": "6",
  "@ionic/vue-router": "6",
```
5. Reload browser and go back to step 1

### Ionic Info

**Ionic**
- Ionic CLI       : 6.18.1
- Ionic Framework : @ionic/vue 6.0.10

**Capacitor**
-    Capacitor CLI      : 3.3.2
-    @capacitor/android : 3.3.2
-    @capacitor/core    : 3.3.2

**Comment by nguyentichduy2k** [comment]
Oh sorry I missed the z at the end of the link, please check it again. [https://codesandbox.io/s/example-router-bug-91320z](url)

**Comment by nguyentichduy2k** [comment]
Thanks for answer but it only works with 2 tabs when 3 tabs it don't work

**Comment by liamdebeasi** [comment]
Thanks, I can reproduce this behavior. The problem is that when going from Tab 3 to the defaultHref Home, Ionic does not find a leaving view. It tries to look for Tab 3, but since the leaving view should be Tabs (since you are leaving tabs), it returns `undefined`.

**Comment by nguyentichduy2k** [comment]
yeah but i still found it fine in ionic 5.9.3 version, there was some change at ionic 6 that generated error

**Comment by gzusgo-thinkbig** [comment]
I been having a similar problem the router found the the exact match but other views are stack hover the right one you can see the right url the wrong view been display. very annoying and is all over the place.  

We are using ver:
"@ionic/vue": "~6.7.5",
"@ionic/vue-router": "~6.7.5",

Not sure we can downgrade our project  to 5.9.3.

**Comment by ShaneK** [comment]
This issue has been resolved at some point over the last 3 years. as I can not reproduce it in 8.8.4 and the reproduction URL doesn't exist anymore. As such, I'm going to go ahead and close this as done, but if you find a way to reproduce this in 8.84, we'd be happy to look into it if you open a new issue for it
